src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -1617,10 +1618,19 @@ private static void sendMessagesToNonlocalDC(Message<? extends IMutation> messag
             target = targets.get(0);
         }
 
+        Tracing.trace("Sending mutation to remote replica {}", target);
         MessagingService.instance().sendWriteWithCallback(message, target, handler);
         logger.trace("Sending message to {}@{}", message.id(), target);
     }
 
+    private static Replica pickReplica(EndpointsForToken targets)
+    {
+        EndpointsForToken healthy = targets.filter(r -> DynamicEndpointSnitch.getSeverity(r.endpoint()) == 0);
```

Comment:
one negative with this is what we keep parsing severity... it may make sense for EndpointState to cache...

Reply:
I think that should be fast enough without caching.

Reply:
cache would require volatile, where as parsing the string to an int is fixed cost/memory... so likely better than the volatile.